app/src/main/java/com/celzero/bravedns/receiver/NotificationActionReceiver.kt

```diff
@@ -69,7 +91,20 @@ class NotificationActionReceiver : BroadcastReceiver() {
     }
 
     private fun stopDnsVpnService(context: Context?) {
+        Log.d(Constants.LOG_TAG, "NotificationActionReceiver: onReceive")
         VpnController.getInstance().stop(context)
     }
 
+    private fun dnsMode(){
+        Log.d(Constants.LOG_TAG, "NotificationActionReceiver: dnsMode")
+        HomeScreenActivity.GlobalVariable.braveMode = HomeScreenFragment.DNS_MODE
+        persistentState.setBraveMode(HomeScreenFragment.DNS_MODE)
+    }
+
+    private fun dnsFirewallMode(){
+        Log.d(Constants.LOG_TAG, "NotificationActionReceiver: dnsFirewallMode")
+        HomeScreenActivity.GlobalVariable.braveMode = HomeScreenFragment.DNS_FIREWALL_MODE
+        persistentState.setBraveMode(HomeScreenFragment.DNS_FIREWALL_MODE)
```

Comment:
Why do we set the persistent state value in a global variable too? Can't the files simply fetch the state from `persistentState`? Please get rid of such redundant states.

Reply:
Removed the global variable from `NotificationActionReceiver`.